lib/assessment.js

```diff
@@ -447,26 +447,39 @@ module.exports = {
           }
 
           const succeeded = !!_.get(content, 'grading.feedback.results.succeeded', true);
-          const gradable = !!_.get(content, 'grading.feedback.results.gradable', true);
+          let gradable = !!_.get(content, 'grading.feedback.results.gradable', true);
           if (!succeeded) {
             content.grading.score = 0;
           }
 
           if (gradable) {
             // We only care about the score if it is gradable.
+            if (typeof content.grading.score === 'undefined') {
+              content.grading.feedback = {
+                results: { succeeded: false, gradable: false },
+                message: 'Error parsing external grading results: score was not provided.',
+                original_feedback: content.grading.feedback,
+              };
+              content.grading.score = 0;
+              gradable = false;
+            }
             if (!_(content.grading.score).isNumber()) {
```

Comment:
Also exclude +/- infinity and NaN:
```suggestion
            if (!_(content.grading.score).isFinite()) {
```